shared/js/background/utils.es6.js

```diff
@@ -185,6 +185,16 @@ function isSafeListed (url) {
     return false
 }
 
+/**
+ * Tests whether the two URL's belong to the same
+ * first party set.
+ */
+function isFirstParty (url1, url2) {
+    const first = tldts.parse(url1).domain
+    const second = tldts.parse(url2).domain
+    return first === second
```

Comment:
hm, I wonder what will happen here for edge cases like IPs and chrome://newtab etc.

cc @sammacbeth

Reply:
Do we need to worry about IP addresses in this implementation? If so is there an easy check if a domain is an IP address so we can compare? Also cc @Charlie-belmer who originally wrote this method for UA rotation.

Reply:
IMO we should be taking IPs into account, tldts returns `isIP` property so that should be straightforward.

As for the private domains, I think we should stick to the standard definition of a third party, as used by other browsers, and don't use private domains for now. It's a good idea for a follow up though.